Yoast/yoast-acf-analysis
# Remove testing code from plugin

## Summary

As noticed in #103 it is not ideal to handle the loading of testing data inside the production plugin code as said in #111.

I've been thinking about several approaches: This PR proposes the one I like the most.

## Relevant technical choices:

* No code left in Plugin itself
* Load dev bootstrap with composers [autoload/files functionality](https://getcomposer.org/doc/04-schema.md#files)
* Use normal composer autoloader with PHP 5.3.2+ as [`xrstf/composer-php52`](https://github.com/composer-php52/composer-php52) apparently doesn't support the `files` autoload functionality, hence this only works with PHP 5.3.2+. But as this is only needed for client-side tests those shouldn't really care about the PHP version in use.
* Adds hooks before WP is initialized by adding a hook to  `global $wp_filter`

## Test instructions

This PR can be tested by following these steps:

* Checkout the branch and make sure to `composer dumpautoload` afterwards so the autoloader is regenerated.

Possibly fixes #111
